### PR TITLE
fix: remove old checkouts on non-hermetic runners

### PR DESCRIPTION
**Pull request by spencerdrak**
Leftover moved submodules were causing issues on non-hermetic runners.

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by emjin — APPROVED** [review]
Nuclear but I see no reason why we shouldn't do it. Do we also need it for linux?

**Comment by spencerdrak** [comment]
@emjin agreed - this is what we're doing in every other case, including the non-hermetic runners in Semgrep Pro as well as any runner that is GH hosted, since we get a new env each time. 

No, linux uses a GH hosted runner, with a fresh env each run.

Once I get a codeowner review I'll merge.

**Review by bmahe — APPROVED** [review]

**Pull request merged**
